numpy/lib/arraypad.py

```diff
@@ -879,6 +879,15 @@ def pad(array, pad_width, mode, **kwargs):
     stat_functions = {"maximum": np.max, "minimum": np.min,
                       "mean": np.mean, "median": np.median}
 
+    if mode == "constant":
+        values = kwargs.get("constant_values", 0)
+        values = _as_pairs(values, array.ndim)
+        if len(set(values)) == 1:
```

Comment:
Even taking the `set` may not be good for object type. I think you would probably need to test for numeric types, maybe through: `array.dtype.kind in "buifc"`.

Reply:
I just tried this with the test suite. Sadly this fails for some tests because `_as_pairs` may return a nested tuple like `((x, y),)` :

```
values = ((1, 2),)
len(set(values)) == 1
```